File: src/main/java/org/springframework/samples/nt4h/card/enemy/inGame/EnemyInGame.java
package org.springframework.samples.nt4h.card.enemy.inGame;

import lombok.*;
import org.springframework.samples.nt4h.card.enemy.Enemy;
import org.springframework.samples.nt4h.model.BaseEntity;

import javax.persistence.*;
import javax.validation.constraints.Max;
import javax.validation.constraints.NotNull;
import java.util.ArrayList;
import java.util.List;


@Getter
@Setter
@Entity
public class EnemyInGame extends BaseEntity {

    public EnemyInGame(Enemy enemy) {
        this.actualHealth = enemy.getHealth();
        List<String> nighLordsNames = List.of("Gurdrug", "Roghkiller", "Shriekknifer");
        this.isNightLord = nighLordsNames.contains(enemy.getName());
        this.enemy = enemy;
    }
    @NotNull
    @Max(value = 10)
    private Integer actualHealth;

    @NotNull
    private boolean isNightLord;

    @ManyToOne
    private Enemy enemy;

    public void onDeleteSetNull() {
        enemy = null;
    }

}

File: src/main/kotlin/org/springframework/samples/nt4h/card/enemy/EnemyInGame.kt
package org.springframework.samples.nt4h.card.enemy

import org.springframework.samples.nt4h.card.decorator.Decorator
import org.springframework.samples.nt4h.model.BaseEntity
import javax.persistence.CascadeType
import javax.persistence.Entity
import javax.persistence.ManyToOne
import javax.persistence.OneToMany
import javax.validation.constraints.Max

@Entity
data class EnemyInGame(
    // Attributes -------------------------------------------------------------
    @Max(value = 10)
    var actualHealth: Int = 0,
    var isNightLord: Boolean = false,
    var isDefeated: Boolean = false,
    var turnCount: Int = 0,

    // Relationships ----------------------------------------------------------
    @ManyToOne(cascade = [CascadeType.ALL])
    var enemy: Enemy? = null,
    @OneToMany(cascade = [CascadeType.ALL])
    var decorators: MutableList<Decorator> = mutableListOf()
) : BaseEntity()
